### PR TITLE
Add SaveFile hook and its test

### PR DESCRIPTION
**Pull request by gdynusa**

### CODE REVIEW

File: emloop/hooks/save_file.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class SaveFile(AbstractHook):
+    """
+    Save files to the output dir before training.
+
+    .. code-block:: yaml
+        :caption: save files to output dir
+
+        hooks:
+          - SaveFile
```
Comment (petrbel) [inline emloop/hooks/save_file.py:19]: this doesn't seem like a valid configuration as `files` parameter has no default value

File: emloop/hooks/save_file.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        :param output_dir: directory to save the files
+        """
+        super().__init__(**kwargs)
+
+        self._files = files
+        self._output_dir = output_dir
+
+    def before_training(self):
+        for path in self._files:
+            assert os.path.exists(path), f'file {path} does not exist'
```
Comment (petrbel) [inline emloop/hooks/save_file.py]: please add "`" (backticks) around the `{path}`. It's easier to debug when you see that there is an empty file etc.

**Review by petrbel — DISMISSED** [review]
Good! Just minor changes pls

**Review by FloopCZ — APPROVED** [review]
Thank you @gdynusa!

**Comment by FloopCZ** [comment]
And thank you @bedapisl as well. :-)

**Pull request merged**